### PR TITLE
feat: vector api

### PR DESCRIPTION
**Pull request by aldarisbm**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,25 +1,23 @@
-module github.com/nekomeowww/go-pinecone
+module github.com/aldarisbm/go-pinecone
```
Comment (nekomeowww) [inline go.mod:1]: module name shouldnt be changed

File: requestvalidators.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	if params.SparseVector != nil && len(params.SparseVector.Values) != len(params.SparseVector.Indices) {
+		return fmt.Errorf("%w: sparse vector values and indices must be the same length", ErrInvalidParams)
+	}
+
+	return nil
+}
+
+// validateDeleteVectorsParams validates the delete vectors parameters.
+func validateDeleteVectorsParams(params DeleteVectorsParams) error {
+	if params.Ids == nil && len(params.Ids) < 1 && !params.DeleteAll {
```
Comment (nekomeowww) [inline requestvalidators.go:28]: it is safe to use `len()` to obtain and determine the length of the slice without `== nil` expression

File: vectors.go [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+// See https://docs.pinecone.io/reference/query for more information.
+type QueryParams struct {
+	Filter          map[string]any `json:"filter"`
+	IncludeValues   bool           `json:"includeValues"`
+	IncludeMetadata bool           `json:"includeMetadata"`
+	Vector          []float32      `json:"vector"`
+	SparseVector    *SparseVector  `json:"sparseVector"`
+	Namespace       string         `json:"namespace"`
+	TopK            int            `json:"topK"`
+	Id              string         `json:"id"`
```
Comment (nekomeowww) [inline vectors.go:63]: All the `Id` field names should be replaced with `ID`

File: vectors_test.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+		})
+
+	})
+}
+
+func TestIndexClient_FetchVectors(t *testing.T) {
+	t.Run("should return error if namespace is empty", func(t *testing.T) {
+		ic := IndexClient{}
+		_, err := ic.FetchVectors(context.Background(), FetchVectorsParams{})
+		if err == nil {
```
Comment (nekomeowww) [inline vectors_test.go:37]: You could replace the err assertion with `require.NoError`

**Comment by aldarisbm** [comment]
I'm rebasing my branch, and adding tests, I didnt mean to open this PR. real PR coming soon, but will take into account your comments

**Pull request closed without merging**